test/distributed/test_c10d_ucc.py

```diff
@@ -1031,19 +1072,61 @@ def test_nested_comm_tensor_wrapping_gpu(self):
             torch.ones(2, 2, device=self.rank) * self.rank
         )
 
-    @skip_if_lt_x_gpu(2)
     def test_consecutive_comm_work_wait_gpu(self):
         self._test_consecutive_comm_work_wait(
             torch.ones(2, 2, device=self.rank) * self.rank
         )
 
+    def test_allreduce_work_wait_cpu(self):
+        self._test_allreduce_work_wait(
+            torch.ones(2, 2) * self.rank,
+        )
+
+    def test_allgather_work_wait_cpu(self):
+        self._test_allgather_work_wait(
+            torch.ones(2, 2) * self.rank
+        )
+
+    def test_broadcast_work_wait_cpu(self):
+        self._test_broadcast_work_wait(
+            torch.ones(2, 2) * self.rank
+        )
+
+    def test_nested_comm_tensor_wrapping_cpu(self):
+        self._test_nested_comm_tensor_wrapping(
+            torch.ones(2, 2) * self.rank
+        )
+
+    def test_consecutive_comm_work_wait_cpu(self):
+        self._test_consecutive_comm_work_wait(
+            torch.ones(2, 2) * self.rank
+        )
+
 
 class UccProcessGroupWithDispatchedCollectivesTests(test_c10d_common.ProcessGroupWithDispatchedCollectivesTests):
+
     @requires_ucc()
     @skip_if_lt_x_gpu(1)
     def test_collectives(self):
+        # includes reduce, broadcast, all_reduce, all_gather, reduce_scatter, barrier, all_to_all, scatter
         self._test_collectives(backend="ucc")
 
+    @requires_ucc()
+    @skip_if_lt_x_gpu(1)
+    def test_allgather_base(self):
+        store = dist.FileStore(self.file_name, self.world_size)
+        dist.init_process_group(
+            "ucc",
+            world_size=self.world_size,
+            rank=self.rank,
+            store=store,
+        )
+        device = "cuda"
```

Comment:
There is no device ID here. Is this based on an assumption that `set_current_device` has been called somewhere before?

Reply:
nccl equivalent of this test uses default cuda device (https://github.com/pytorch/pytorch/blob/main/test/distributed/test_c10d_nccl.py#L2995), so I just followed the convention. Doesn't seem like cuda.set_device is used.

Reply:
So this test just tests for correct dispatching to device backend.
And the base test class assumes a world_size of 1:
https://github.com/pytorch/pytorch/blob/main/test/distributed/test_c10d_common.py#L1692-L1695